sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/CheckpointManager.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.messaging.eventhubs.eventprocessor.models.Checkpoint;
+import com.azure.messaging.eventhubs.eventprocessor.models.PartitionContext;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * The checkpoint manager that clients should use to update checkpoints to track progress
+ * of events processed
+ */
+public class CheckpointManager {
+    private PartitionContext partitionContext;
```

Comment:
`private final` for partitionContext and partitionManager to explicitly tell people you don't' want them reassigning this variable.